packages/api/session.mts

```diff
@@ -296,7 +298,69 @@ export function updateCell(session: SessionType, cell: CellType, updates: CellUp
       return updateCodeCell(session, cell, updates);
   }
 }
-
+async function ensurePrettierInstalled(dir: string): Promise<boolean> {
```

Comment:
Some thoughts here:

1. If formatting is going to be a first class citizen in all Srcbooks, then we should create every new srcbook with it already present in the package.json's `devDependencies`
2. If we do check for its presence here, are we able to use [`depcheck`](https://github.com/depcheck/depcheck)? We already use this in `deps.ts` and it looks like it has "special" support for prettier.
3. If the dep is missing, I'm inclined to prompt the user for installing it. While this does introduce an extra manual step, it keeps the user in the loop and gives faster feedback than if they waited for install to happen behind the scenes. Most importantly, this should rarely happen if we include prettier in the dependencies when creating new srcbooks.

Reply:
I stand behind number 3 above but in the interest of getting this out sooner than later, we can ignore that piece for now and leave this code for now (though would be good to use depcheck).

Reply:
Ohh I just noticed we are doing #1 here already, nice